### ISSUE TITLE
Collection runner - After run, "Data unavailable" for response headers and body

### ISSUE THREAD

**Issue by Ocsidot** [a85/POSTMan-Chrome-Extension#4296]
## App Details:
Postman for Windows
Version 6.0.9
win32 6.1.7601 / x64

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there: This issue occurred recently, after last update
 2. Expected behaviour: In previous version, we can get information on every request in a collection after a run in collection runner.
 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):
 4. Screenshots (if applicable)

![2018-03-07 17_52_01-collection runner](https://user-images.githubusercontent.com/6094145/37105827-720f278e-2230-11e8-8f01-28dd6bd7b762.png)

Since the last update, I cannot get any information in collection runner for any request

**Comment by sivcan** [comment]
@Ocsidot - Thanks for reporting this.
You can expect a fix for this in the upcoming releases for the app.
I'll update this thread once it's available.

**Comment by Ocsidot** [comment]
Thank you very much @sivcan :).
I'll be waiting for the fix

**Comment by sivcan** [comment]
Hi @Ocsidot - The new update has been released (6.0.10) and it has the fix for the same.
Closing this issue.

**Comment by ddeshpande** [comment]
Hi @sivcan, I am seeing this issue in 6.0.10 version for Response Body:

![image](https://user-images.githubusercontent.com/25217651/38128607-e34515e0-3418-11e8-8978-29216d91e850.png)

**Comment by sivcan** [comment]
@ddeshpande - I guess these are the runs that were ran in the earlier version.
The response was never saved for those thus you're seeing this error.

Can you run a collection in the updated version and let me know if this issue still persists for you ?

**Comment by ddeshpande** [comment]
@sivcan I ran my collection in updated version (6.0.10) and observed this issue. Checked it again today:
![image](https://user-images.githubusercontent.com/25217651/38188486-d299ab94-3679-11e8-9d31-2ce6adebf3ee.png)

**Comment by ddeshpande** [comment]
It started working! 

Looks like this was happening because I had sync issues. If you see the screenshot in my previous comment, you will see postman in "Syncing" state. I was being prompted multiple times to "Resync" a collection (which by the way was a different collection, and not the one which I was running in runner). Even after selecting "Resync" multiple times, it was not able to successfully sync.

I restarted Postman app and performed "Resync" again and it successfully synced. Now I am not seeing "Data Unavailable" issue in Runner.

**Comment by sivcan** [comment]
@ddeshpande - Oh great! 

Let me know if you experience this issue again (do check the console logs in DevTools and share the same if you see any error).

Thanks

**Comment by ddeshpande** [comment]
hi @sivcan I'm seeing this issue again! I don't see any error in console logs in DevTools. 
Note: The sync issue is not seen anymore, Postman is "In Sync". But this "Data Unavailable" issue has popped up again.

Thanks,
Deepti

**Comment by ddeshpande** [comment]
Could it be because I am using multiple workspaces? After I updated Postman to latest version 6.0.10, I created new workspaces and organized my collections in these workspaces..

**Comment by sivcan** [comment]
Hey, @ddeshpande - If your data is syncing then unfortunately the responses are not synced. 
You can view them on your local machine. 
If you're logging in / out of the app or using multiple desktops then you won't be able to view the responses. 
Switching to multiple workspaces will not be having this issue. 

I'll still check this and get back to you.

**Comment by ddeshpande** [comment]
Hi @sivcan, any update on this issue? It's still reproducible for me. Here's what I have observed:

1. I run a request in postman, it runs successfully. I get the response, my tests have passed and global variables that I have marked to be set upon getting response, get set successfully.
2.  If I run the same request in Runner, it runs successfully, I can see the response and all my tests have passed. But the global variables don't get set.
3. I click on "Retry" button, it runs successfully again and my tests are again passed. But now the response is not shown, instead I see - "Data unavailable (Only data about the top 10 historical runs is stored)" and again global variables are not set. 

As variables are not getting set, I am not able to run my flows. Please check if the below console logs are of any help:

![image](https://user-images.githubusercontent.com/25217651/38483257-ec8c6e76-3bef-11e8-9f60-797cda89cf12.png)

It would be great if this can be fixed soon. Thanks!

Note: As a workaround, I am running my collections in newman. It works fine in newman.

**Comment by sivcan** [comment]
@ddeshpande - Apologies for the delay. 
The switching workspaces had no affect on responses.

Thanks for reporting the other issue.
I was able to reproduce this issue on retrying the run that the data responses are not being stored.
I'll reopen this issue and mark this as a bug.

Will update this thread once the fix is out for the same.

**Comment by ddeshpande** [comment]
Thanks @sivcan !

**Comment by amitbpal** [comment]
@sivcan I am facing the same problem.

My postman version is -6.1.4 (6.1.4)
Please help me out.

**Comment by SkyGanfor** [comment]
@amitbpal I was just having the same issue. I updated postman, did a postman reboot, and that solved it for me.

EDIT: The issue happened to me again. A reboot of postman seems to have solved it temporarily.

**Comment by amitbpal** [comment]
@SkyGanfor  thanks for your feedback . Can you please let me know how I can reboot my postman application ??

**Comment by SkyGanfor** [comment]
@amitbpal Just close the postman program and open it again. Probably should have used better wording than reboot :)

**Comment by amitbpal** [comment]
@SkyGanfor I did this many times but no luck :(

**Comment by amitbpal** [comment]
@SkyGanfor  is there any other way that I can run my collection again . :(

It's very urgent. please help me.

**Comment by SkyGanfor** [comment]
@amitbpal Maybe a reinstall would work? That's all I can think of :(

**Comment by sivcan** [comment]
@amitbpal @SkyGanfor What exactly is the message showing?
Does the message state 'Data unavailable' / 'Response body is too large'?
Also, are you retrying the run? If yes, then we're aware of this issue and I'll update the thread once a fix is out.

Can you guys tell us the steps to reproduce this issue on our end (except retrying the run)?

**Comment by amitbpal** [comment]
@sivcan I was trying to run my collection with the multiple API calls before it was working for me but from last 3-4 days I am not able to run my test from the runner.

I am getting "Data unavailable(only data about the top 10 historical runs is stored)" error message in the response body.

**Comment by sivcan** [comment]
@amitbpal 
Couple of questions that would help us to triage this issue faster:

1. So you're not **retrying** the run right? (Since you didn't confirm this before so asking again)
This just started happening on running every run or a few old ones? 
2. Are these runs being synced and you're opening them on another machine?
3. Are you facing this issue while running the run or after closing the runner and opening it back again?

**Comment by amitbpal** [comment]
@sivcan here are your questions answer below : 

1. On Every run .
2.They are syned and I am using same machine.
3.I have close the runner many times and try to run again but no luck for me.

**Comment by michael-upton97** [comment]
I know that no fix has been announced yet, but I am still experiencing this issue. Often returning to the Collection Runner primary tab and running the tests from the blue button will allow for the response body to be visible, However this doesn't always work, and occasionally I can still get a response body when using the retry button. 

Can we be expecting a fix soon or is this not something that is being addressed?

P.S. Using Postman Version 6.3.0

Update, reinstalled with a 64bit version and was able to update to version 6.4.1, still experiencing same issue as far as I can tell.

![image](https://user-images.githubusercontent.com/35309613/46592531-4c0c3980-ca89-11e8-87cd-3f27fc0286ce.png)

**Comment by allyn-psyonix** [comment]
I was seeing this issue in 6.3, but just updated to 6.4.2 and it went away.

UPDATE: Still experiencing the issue, just seems to be intermittent. Saw it again on 6.4.2 and now on 6.5.2 as well.

**Comment by vandana7** [comment]
I was also facing the same issue but clearing the recent Runs works for me as If there are many runs through the collection runner it only stores the recent runs Data. (deleting recent runs is a temp fix :P)

**Comment by fsebile** [comment]
I'm still facing this issue here 👎 I really need this feature and tried previous versions, same bug. Any recommendation for solution didn't worked for me. Here my screenshots.
![screenshot_1](https://user-images.githubusercontent.com/5302483/52614278-3dec8080-2ea2-11e9-996c-da68a43354ec.png)
![image](https://user-images.githubusercontent.com/5302483/52614297-4d6bc980-2ea2-11e9-9b83-a9f9d6919ecc.png)

**Comment by klemens-u** [comment]
I have the same problem with Postman v7.0.9.

I just run a collection and wanted to debug a failed request. 

![image](https://user-images.githubusercontent.com/1863743/56665277-05d79a00-66aa-11e9-9271-10267917abc0.png)

What does "top 10" mean? I understand it can be a resource problem when so much debug information is stored. But to store data for the latest run is absolutely essential!

**Comment by christhomas** [comment]
The fact that they tell you only the top 10 are stored means that this can't be a bug, it's programmed in behaviour. Otherwise it wouldn't say specifically 10 historical run is stored.

But what means a historical run? I get this on the first run, the last run shouldn't count as being more than the last 10 historical runs, right?

Or does it mean it only stores response bodies for the last 10 requests. Meaning if I have 100 tests the data from the first 90 are deleted? However this is a pretty obvious failure. How can this be useful if I can't see the responses from the previous run? I'm not sure how it is supposed to work but that doesn't sound very useful behaviour to me.

**Comment by craigmcnally** [comment]
Still an issue in 7.2.2 :(

**Comment by schadsey** [comment]
Still an issue in 7.3.3.  However, it seems to occur on the 2nd and all subsequent run of a collection within the same Collection Runner session (i.e., clicking "Retry" after the 1st run).  If I close the runner and re-open it and run the collection, I see the response headers and bodies.  So the workaround appears to be to close and re-open the Runner window if you want to be able to examine the response details for each request.

**Comment by sarangjaiswal** [comment]
This issue exist in 7.10.0 as well. Do you have ETA by when this will be fixed?

**Comment by tomprogers** [comment]
It's really hard to take Postman seriously as a testing tool when the debugging experience is so crummy.

1. It is unacceptable for the test runner to discard any information from the test run. All of it might  be needed to diagnose failures. It should not matter which UI button the dev uses to trigger the run. All this nonsense about "are you retrying vs running from scratch" is beside the point. It must work the same way -- i.e. *correctly* -- in all cases or it cannot be relied upon.
2. None of this should depend on syncing. This data must be available immediately on the machine where the test were run, regardless of any additional sharing/publishing functionality. It is pointless to share data when that data is inadequate to diagnose test failures.

This is a big hurdle that was reported in the previous major version, and now it's 18 months later and the problem still persists.

It's impossible to take Postman seriously as a testing tool when it fails developers so badly in the very scenarios where testing is vital.

If I were you, I wouldn't bother releasing any other features until this is fixed. No other feature could compensate for these problems (unless it's a complete overhaul of how testing works).

**Comment by bigbigmo** [comment]
I have the same issue. It's a terrible situation...

**Comment by ghost** [comment]
Still an issue. Very important for resolving test issues

**Comment by klemens-u** [comment]
Please fix this. Very annoying. Otherwise great product.

**Comment by christhomas** [comment]
Tomprogers says exactly what we all are thinking. How can this be a serious TESTING tool, when it's unreliable for it's most basic TESTING functionality.

To quickly knock up some API requests and execute them, Postman is great. But this whole testing stuff means you lost me as a user for this functionality until something is done about it

**Comment by allyn-psyonix** [comment]
This has been a constant issue and was particularly frustrating when trying to debug an issue that was only appearing in a CI environment today. It's been broken for quite a while (issue has been open for a year and a half). Is there any update on this?

The error I received today was slightly different than those shown above:

> Response body was not logged

**Comment by saswatds** [comment]
Hi everyone we have been working hard at providing a more deterministic experience around response headers and body. During our research, we found that constant accumulation of all the response was causing the app to crash for long-running collection runs. 

*Starting from app version `7.16.0` you will see*

We have introduced a setting `Save responses` to allow saving for responses up to 300KB for each request. This setting is by default turned off and you will see the message `Response body was not logged`. If the response body exceeds 300KB you will see the message `Response body is too large` 
<img width="530" alt="Screenshot 2020-02-12 at 1 20 23 PM" src="https://user-images.githubusercontent.com/8796447/74313870-7dda9200-4d9a-11ea-9b54-4cc3f9d0e563.png">

The 300KB limit does not apply to the `Postman Console` so you will still be able to see responses up 5MB on the postman console and it works even if you have turned `Save responses` to OFF.

Note: At this moment we do not support syncing of responses for collection runs, so in cases where you sign out and sign back in and the runs are synced then the responses will not be preserved and you will see the message `Response body unavailable`.
